### PR TITLE
Added Scrape builder and extended provided data

### PR DESCRIPTION
**Pull request by nightbloos**

### CODE REVIEW

File: goscraper.go [hunk 1]
```diff
@@ -66,6 +128,23 @@ func (scraper *Scraper) Scrape() (*Document, error) {
 	return doc, nil
 }
 
+func (scraper *Scraper) GetDocument() (*Document, error) {
+	doc, err := scraper.getDocument()
```
Comment (vhbit) [inline goscraper.go]: is it required to have this error dance in here or we can just `return scraper.getDocument()`?

File: goscraper.go [hunk 2]
```diff
@@ -66,6 +128,23 @@ func (scraper *Scraper) Scrape() (*Document, error) {
...
+	if err != nil {
+		return nil, err
+	}
+
+	return doc, nil
+}
+
+func (scraper *Scraper) ParseDocument(doc *Document) (*Document, error) {
+	err := scraper.parseDocument(doc)
```
Comment (vhbit) [inline goscraper.go:136]: same here, maybe `return scraper.parseDocument` will be an easier way to handle it?

File: goscraper.go [hunk 3]
```diff
@@ -19,6 +19,63 @@ var (
...
+	SetUrl(string) ScrapeBuilder
+	SetMaxRedirect(int) ScrapeBuilder
+	GetScraper() (*Scraper, error)
+}
+
+type scrapeBuilder struct {
+	scrapeSettings scrapeSettings
+}
+
+func (b *scrapeBuilder) GetScraper() (*Scraper, error) {
```
Comment (vhbit) [inline goscraper.go]: I don't know what is more common in `go` land - to have `GetXXX` from `XXXBuilder` or to have a generic `Build`, please check in other libraries

Comment (nightbloos) [reply goscraper.go]: In some places find that is more used `Build`.

**Pull request merged**